v2/yarpcthrift/thriftrw-plugin-yarpcv2/client.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"path/filepath"
+
+	"go.uber.org/thriftrw/plugin"
+)
+
+const clientTemplate = `
+// Code generated by thriftrw-plugin-yarpc
+// @generated
+
+<$pkgname := printf "%sclient" (lower .Name)>
+package <$pkgname>
+
+<$yarpc     := import "go.uber.org/yarpc/v2">
+<$yarpcthrift    := import "go.uber.org/yarpc/v2/yarpcthrift">
+
+</* Note that we import things like "context" inside loops rather than at the
+    top-level because they will end up unused if the service does not have any
+    functions.
+ */>
+
+// Interface is a client for the <.Name> service.
+type Interface interface {
+	<if .Parent><import .ParentClientPackagePath>.Interface
+	<end>
+	<range .Functions><if not .OneWay>
+		<$context := import "context">
+		<.Name>(
+			ctx <$context>.Context, <range .Arguments>
+			<.Name> <formatType .Type>,<end>
+			opts ...<$yarpc>.CallOption,
+		)<if .ReturnType> (<formatType .ReturnType>, error)
+		<else> error
+		<end>
+	<end><end>
+}
+
+</* TODO(abg): Pull the default routing name from a Thrift annotation? */>
```

Comment:
Let’s capture this TODO in CHANGES. Maybe we make a ticket later. This has been lost to the annals of time.